tck/src/main/java/org/reactivestreams/tck/SubscriberBlackboxVerification.java

```diff
@@ -231,6 +234,77 @@ public String toString() {
+      @Override
+      public void run(BlackboxTestStage stage) throws Throwable {
+        final Subscriber<T> sub = createSubscriber();
+
+        final AtomicInteger cancellationCounter = new AtomicInteger(0);
+        final AtomicReference<Subscription> firstAcceptedSubscription = new AtomicReference<Subscription>(null);
+        final AtomicReference<Subscription> illegallyAcceptedSubscription = new AtomicReference<Subscription>(null);
+
+        final int tasksToSubmit = 8 * Runtime.getRuntime().availableProcessors();
+
+        Callable<Void> invokeOnSubscribe = new Callable<Void>() {
+          final Subscriber<? super T> targetSubscriber = sub;
+
+          @Override
+          public Void call() throws Exception {
+            targetSubscriber.onSubscribe(new Subscription() {
+
+              // we keep around the thread name to help identify which thread's subscription was accepted first,
+              // and then illegally second
+              final String nameWithThread = "Subscription<concurrently issued on thread " + Thread.currentThread().getName() + ">";
+
+              @Override public void request(long elements) {
+                if (firstAcceptedSubscription.compareAndSet(null, this)) {
+                  env.debug(String.format("Accepted subscription %s (first)", firstAcceptedSubscription.get()));
+                  // first one accepted, this is legal
+                  // <empty on purpose, the effect we need has been achieved by the CAS above)
+                } else {
+                  // seems two subscriptions were issued a request() signal concurrently;
+                  // this is not legal, since only ONE should have been accepted; and the other one cancelled.
+                  env.flop(String.format("Subscriber %s ilegally accepted and issued request() to two subscriptions, violating rule 2.5; " +
+                      "The accepted subscriptions are: %s and %s", sub, this, firstAcceptedSubscription.get()));
+                }
+              }
+
+              @Override public void cancel() {
+                int cancelledCounter = cancellationCounter.incrementAndGet();
+                env.debug(String.format("Subscription %s rejected; All but one subscriptions are expected to be rejected here; " +
+                    "Already rejected subscriptions: %d out of %d total subscriptions.", this, cancelledCounter, tasksToSubmit));
+              }
+
+              @Override public String toString() { return nameWithThread; }
+            });
+            return null;
+          }
+        };
+
+        List<Callable<Void>> invokeOnSubscribeConcurrentlyTasks = new ArrayList<Callable<Void>>(tasksToSubmit);
+        for (int i = 0; i < tasksToSubmit; i++) {
+          invokeOnSubscribeConcurrentlyTasks.add(invokeOnSubscribe);
+        }
+
+        List<Future<Void>> futures = Executors.newFixedThreadPool(16).invokeAll(invokeOnSubscribeConcurrentlyTasks);
```

Comment:
Shouldn't this Executor shut down when no longer needed? Also having more threads than cores may actually reduce the chance of hitting the desired concurrency effect because of the increased context switching among threads.

Reply:
Yes it should; first let's agree if this PR makes sense or not -- as we started in the ticket, if not, no reason to spend time polishing it ;-)